src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesigntimeLicenseContextSerializer.cs

```diff
@@ -25,25 +24,97 @@ private DesigntimeLicenseContextSerializer()
         /// </summary>
         public static void Serialize(Stream o, string cryptoKey, DesigntimeLicenseContext context)
         {
-            IFormatter formatter = new BinaryFormatter();
+            if (LocalAppContextSwitches.BinaryFormatterEnabled)
```

Comment:
Instead of this why not always write the new format and allow folks to control writing old format with a different switch?

Reply:
Right, imagine a complex app where for compat reasons they do need to allow binary formatter somewhere else, they may not want to simultaneously make this use BF too.

Reply:
Ah, good idea. Renamed the switch to `System.ComponentModel.TypeConverter.EnableUnsafeBinaryFormatterSerialization` now so it's decoupled from `"System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization"`

Reply:
I'm not sure about the naming of the switches though given the last word is the same. Looking for feedback here.